cmd/helm/search_repo.go

```diff
@@ -159,7 +159,21 @@ func (o *searchRepoOptions) applyConstraint(res []*search.Result) ([]*search.Res
 			continue
 		}
 		v, err := semver.NewVersion(r.Chart.Version)
-		if err != nil || constraint.Check(v) {
+
+		if err != nil {
+			if err != semver.ErrSegmentStartsZero {
+				return res, errors.Wrap(err, "an invalid semver version")
```

Comment:
In search we do not error out on a bad version. This is because there are two different roles involved. Those who create and share charts in one role and those who search a repository to install charts in another role. Many times those who search have no relation to those who create and serve charts. So, if someone gets this error it's not something they can even take an action to fix.

We want that helm user experience to be as easy as possible.

An error should happen in `helm lint` or `helm package` because those tools are used by those who create charts.

The PR for this issue should silently skip charts that have an issue with the verison

Reply:
Thanks for the complete comments, I very much agree with you

Has been fixed in the way you recommend